### ISSUE TITLE
fatal: Not a git repository (or any of the parent directories): .git

### ISSUE THREAD

**Issue by yakir12** [tpapp/skeleton.jl#11]
It seems to work, but that fatal message seems ominous...

```bash
$ julia /home/yakir/skeleton.jl/skeleton.jl Example
fatal: Not a git repository (or any of the parent directories): .git
┌ Info: skeleton.jl
│   Julia_VERSION = v"1.0.1"
└   githash = "(not a repo)"
┌ Info: copy and substitute
│   srcdir = "/home/yakir/skeleton.jl/template"
│   destdir = "Example"
│   replacements =
│    5-element Array{Pair{String,Any},1}:
│          "{UUID}" => UUID("7876af07-990d-54b4-ab0e-23690620f79a")
│       "{PKGNAME}" => "Example"                                   
│        "{GHUSER}" => "yakir12"                                   
│      "{USERNAME}" => "yakir12"                                   
└     "{USEREMAIL}" => "12.yakir@gmail.com"                        
/home/yakir/skeleton.jl/template/.gitignore  =>  Example/.gitignore
/home/yakir/skeleton.jl/template/.travis.yml  =>  Example/.travis.yml
/home/yakir/skeleton.jl/template/LICENSE.md  =>  Example/LICENSE.md
/home/yakir/skeleton.jl/template/Project.toml  =>  Example/Project.toml
/home/yakir/skeleton.jl/template/README.md  =>  Example/README.md
/home/yakir/skeleton.jl/template/docs/Project.toml  =>  Example/docs/Project.toml
/home/yakir/skeleton.jl/template/docs/make.jl  =>  Example/docs/make.jl
/home/yakir/skeleton.jl/template/docs/src/index.md  =>  Example/docs/src/index.md
/home/yakir/skeleton.jl/template/src/{PKGNAME}.jl  =>  Example/src/Example.jl
/home/yakir/skeleton.jl/template/test/runtests.jl  =>  Example/test/runtests.jl
/home/yakir/skeleton.jl/template/test/coverage/Project.toml  =>  Example/test/coverage/Project.toml
/home/yakir/skeleton.jl/template/test/coverage/coverage.jl  =>  Example/test/coverage/coverage.jl
[ Info: git init
Initialized empty Git repository in /home/yakir/tmp/Example/.git/
[ Info: adding documenter (completing the Manifest.toml for docs)
  Updating registry at `~/.julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
 Resolving package versions...
  Updating `~/tmp/Example/docs/Project.toml`
  [e30172f5] + Documenter v0.19.6
  Updating `~/tmp/Example/docs/Manifest.toml`
  [34da2185] + Compat v1.2.0
  [ffbed154] + DocStringExtensions v0.5.0
  [e30172f5] + Documenter v0.19.6
  [2a0f44e3] + Base64 
  [ade2ca70] + Dates 
  [8bb1440f] + DelimitedFiles 
  [8ba89e20] + Distributed 
  [b77e0a4c] + InteractiveUtils 
  [76f85450] + LibGit2 
  [8f399da3] + Libdl 
  [37e2e46d] + LinearAlgebra 
  [56ddb016] + Logging 
  [d6f4376e] + Markdown 
  [a63ad114] + Mmap 
  [44cfe95a] + Pkg 
  [de0858da] + Printf 
  [3fa0cd96] + REPL 
  [9a3f8284] + Random 
  [ea8e919c] + SHA 
  [9e88b42a] + Serialization 
  [1a1011a3] + SharedArrays 
  [6462fe0b] + Sockets 
  [2f01184e] + SparseArrays 
  [10745b16] + Statistics 
  [8dfed614] + Test 
  [cf7118a7] + UUIDs 
  [4ec0a83e] + Unicode 
```

**Comment by tpapp** [comment]
Probably coming from [this line](https://github.com/tpapp/skeleton.jl/blob/48e38f75f898b9c8f53416751e82f057ccd66dff/skeleton.jl#L9). Did you clone the repo, or extract from a tarball? I could suppress this.

**Comment by yakir12** [comment]
cloned.

**Comment by tpapp** [comment]
can you please

1. send me the output of `ls -a /home/yakir/skeleton.jl` (from the shell),
2. put a `@show @__DIR__` and the top of the script and send me the output from the same command line invocation?

**Comment by yakir12** [comment]
1. 
```bash
yakir@debian:~$ ls -a /home/yakir/skeleton.jl
.  ..  .git  Project.toml  README.md  skeleton.jl  template  test  .travis.yml
```
2. From what I understood, I pasted `@show @__DIR__` at line #1 in `/home/yakir/skeleton.jl/skeleton.jl` and ran the following:
```bash
yakir@debian:~/tmp$ julia /home/yakir/skeleton.jl/skeleton.jl Example
#= /home/yakir/skeleton.jl/skeleton.jl:1 =# @__DIR__() = "/home/yakir/skeleton.jl"
fatal: Not a git repository (or any of the parent directories): .git
┌ Info: skeleton.jl
│   Julia_VERSION = v"1.0.1"
└   githash = "(not a repo)"
[...]
```

**Comment by tpapp** [comment]
The plot thickens... what does
```sh
git status /home/yakir/skeleton.jl
```
(from the shell) say?

**Comment by yakir12** [comment]
```bash
yakir@debian:~$ git status /home/yakir/skeleton.jl
fatal: Not a git repository (or any of the parent directories): .git
```
but:
```bash
yakir@debian:~$ cd skeleton.jl/
yakir@debian:~/skeleton.jl$ git status
On branch master
Your branch is up-to-date with 'origin/master'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   skeleton.jl

no changes added to commit (use "git add" and/or "git commit -a")
```
The modification is just me adding that `@show @__DIR__`.

**Comment by tpapp** [comment]
Can you please send me the shell output of
```
git status -- /home/yakir/skeleton.jl
git version
```
?

**Comment by yakir12** [comment]
```bash
yakir@debian:~$ git status -- /home/yakir/skeleton.jl
fatal: Not a git repository (or any of the parent directories): .git
yakir@debian:~$ git version
git version 2.11.0
```

**Comment by tpapp** [comment]
Close to giving up :wink:, but please try
```
git -C /home/yakir/skeleton.jl status
```

**Comment by yakir12** [comment]
```bash
On branch master
Your branch is up-to-date with 'origin/master'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   skeleton.jl

no changes added to commit (use "git add" and/or "git commit -a")
```
Let me know if I should just delete that repo and re-clone.

**Comment by tpapp** [comment]
In theory it is fixed now on `master`, you can discard your changes and pull, or reclone.

**Comment by tpapp** [comment]
f the bug persists, please comment here. Thanks for your help, these things are tricky to track down.

**Comment by yakir12** [comment]
Wow. So I just recloned and still:
```bash
yakir@debian:~/tmp$ julia /home/yakir/skeleton.jl/skeleton.jl Example
fatal: Not a git repository (or any of the parent directories): .git
┌ Info: skeleton.jl
│   Julia_VERSION = v"1.0.1"
└   githash = "(not a repo)"
```

**Comment by tpapp** [comment]
I am somewhat puzzled by this, and can't replicate. But if you investigate and make a PR, that would help a lot.

Basically the only question that needs to be checked is whether the line querying the log from `git` is doing the right think, in the right directory.

**Comment by yakir12** [comment]
See PR #16, it must be some weird thing with calling `@__DIR__` inside a command?

**Comment by tpapp** [comment]
Closed by #16.
